src/server/command/worldedit/mask.zig

```diff
@@ -0,0 +1,58 @@
+const std = @import("std");
+
+const main = @import("main");
+const User = main.server.User;
+
+const Block = main.blocks.Block;
+const Blueprint = main.blueprint.Blueprint;
+const Mask = main.blueprint.Mask;
+
+pub const description = "Set global or per brush edit mask. When used with no mask expression it will clear current mask.";
+pub const usage =
+	\\/mask brush <mask>
+	\\/mask brush
+	\\/mask global <mask>
+	\\/mask global
+;
+
+const SubCommands = enum {brush, global};
+
+pub fn execute(args: []const u8, source: *User) void {
+	var split = std.mem.splitScalar(u8, args, ' ');
+	const first = split.first();
+	switch(std.meta.stringToEnum(SubCommands, first) orelse {
+		source.sendMessage("#ff0000Unknown subcommand: '{s}'", .{first});
+		return;
+	}) {
+		.brush => brush(split.rest(), source),
+		.global => global(split.rest(), source),
+	}
+}
+
+pub fn brush(args: []const u8, source: *User) void {
```

Comment:
How exactly does the brush mask work? Would it be per brush? How would that work?
Otherwise I don't understand why it would need to be a separate thing.

Reply:
For brush masks to make any sense they must be per brush and they should stick same way the shape and patter does. Basically brush is defined by those 3 things. That's how brushes work in WorldEdit and they are very useful. However it may make more sense to have a separate brush command that aggregates those 3 things together instead of scattering them across multiple commands. So `/brush shape` `brush pattern` and `/brush mask` and just single `/mask` with no subcommands for global masks (WE has `/gmask` and just `/mask` for brush masks)

Reply:
Yeah, I think that belongs to the brush system and not here.
Also why not just make it all in one `/brush <shape> <pattern> <mask>`?

Reply:
Because it is often useful to tweak brush as you go, or completely skip certain components. Also patterns can get **really** long, so they will legitimately limit each other.

Reply:
Alright, then please move the brush mask to the brush command and remove it from here.

Reply:
Well, done, except brush mask was moved at conceptual level, since there is no brush command yet.